core/src/main/java/software/amazon/awssdk/auth/AbstractAwsSigner.java

```diff
@@ -224,32 +220,24 @@ protected String getCanonicalizedQueryString(Map<String, List<String>> parameter
          * encoding in addition to sorted parameter names.
          */
         for (Map.Entry<String, List<String>> entry : parameters.entrySet()) {
-            final String encodedParamName = SdkHttpUtils.urlEncode(
-                    entry.getKey(), false);
+            final String encodedParamName = SdkHttpUtils.urlEncode(entry.getKey());
             final List<String> paramValues = entry.getValue();
-            final List<String> encodedValues = new ArrayList<String>(
-                    paramValues.size());
+            final List<String> encodedValues = new ArrayList<>(paramValues.size());
             for (String value : paramValues) {
-                encodedValues.add(SdkHttpUtils.urlEncode(value, false));
+                String encodedValue = SdkHttpUtils.urlEncode(value);
+
+                // Null values should be treated as empty for the purposes of signing, not missing.
```

Comment:
This is the existing behavior correct? Looking at the Sigv4 spec this is correct but just making sure we do it right in V1 as well.

Reply:
Yeah, we do it right in V1. We had a few ways of doing encoding in V1, and the one this one used encoded it in that way. I made all of the encoding use the same path with this, so I had to make the tweak here.